### PR TITLE
Add overview.rb file to Ruby docs

### PR DESCRIPTION
**Pull request by geigerj**
This overview serves as the landing page for documentation for
GAPIC-only APIs. Previously, it was not rendered due to the same
feature shortcoming in the jsondoc tool that motivated

  https://github.com/googleapis/toolkit/pull/1544

Retain readme text in the version index to serve as a per-version
code level example, but remove installation instructions, since
installation is not API version-dependent.

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java [hunk 1]
```diff
@@ -74,6 +76,8 @@ private ViewModel generateDoc(ProtoFile file, GapicProductConfig productConfig)
             new RubyTypeTable(productConfig.getPackageName()),
             new RubyModelTypeNameConverter(productConfig.getPackageName()));
     // Use file path for package name to get file-specific package instead of package for the API.
+    // TODO(jgeiger): switch to use productConfig.getPackageName() instead of
```
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java]: Why do we need to switch?

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java [hunk 2]
```diff
@@ -85,7 +89,23 @@ private ViewModel generateDoc(ProtoFile file, GapicProductConfig productConfig)
...
     doc.elementDocs(elementDocTransformer.generateElementDocs(typeTable, namer, file));
     doc.modules(
-        generateModuleViews(file.getModel(), productConfig, namer, isSourceApiInterfaceFile(file)));
+        generateModuleViews(
+            file.getModel(), productConfig, namer, isSourceApiInterfaceFile(file), false));
+    return doc.build();
+  }
+
+  private ViewModel generateOverview(Model model, GapicProductConfig productConfig) {
+    // TODO(jgeiger): switch to use productConfig.getPackageName() instead of
```
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java]: This namer already uses `productConfig.getPackageName()`

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java (hunk 1 above)
Comment (geigerj) [reply src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java]: Whoops, bad comment. Removed.

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java (hunk 2 above)
Comment (geigerj) [reply src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java]: Removed.

**Pull request merged**